bodo/io/snowflake_write.py

```diff
@@ -50,13 +50,27 @@
 )
 
 
-class SnowflakeWriterType(types.Type):
+class SnowflakeWriterType(StreamingStateType):
     """Data type for streaming Snowflake writer's internal state"""
 
     def __init__(self, input_table_type=types.unknown):
         self.input_table_type = input_table_type
         super().__init__(name=f"SnowflakeWriterType({input_table_type})")
 
+    def is_precise(self):
+        return self.input_table_type != types.unknown
+
+    def unify(self, typingctx, other):
+        """Unify two SnowflakeWriterType instances when one doesn't have a resolved
+        input_table_type.
+        """
+        if isinstance(other, SnowflakeWriterType):
+            if not other.is_precise() and self.is_precise():
+                return self
+
+            # Prefer the new type in case write append changed its table type
+            return other
+
```

Comment:
Duplicate code as below?  Move to StreamingStateType?

Reply:
I feel like it's safer to have it explicitly here and a few lines of code duplication is worth it for this critical codes.